src/core.js

```diff
@@ -489,10 +489,16 @@ $.extend( $.validator, {
 			this.submitted = {};
 			this.prepareForm();
 			this.hideErrors();
-			var i, elements = this.elements()
+			var elements = this.elements()
 				.removeData( "previousValue" )
 				.removeAttr( "aria-invalid" );
 
+			this.resetElements( elements );
+		},
+
+		resetElements: function( elements ) {
```

Comment:
I guess we should move this method somewhere that it is not part of the published api.